Auth_phpbb.php

```diff
@@ -504,15 +590,50 @@ public function getCanonicalName( $username )
 
         // Query Database.
         $fresStatement = $fresMySQLConnection->prepare($fstrMySQLQuery);
-        $fresStatement->bind_param('s', $username);
+        $fresStatement->bind_param('s', $this->_phpBBUserName); // bind_param escapes the string
         $fresStatement->execute();
 
         // Bind result
-        $fresStatement->bind_result($resultUsernameClean);
+        $fresStatement->bind_result($resultWikiUsername);
 
         while($fresStatement->fetch())
         {
-            return ucfirst($resultUsernameClean);
+            $this->_wikiUserName = ucfirst($resultWikiUsername); // Preserve capped phpBB username when wikified version is valid
+            return $this->_wikiUserName;
+        }
+
+        // If here, username is invalid or is incompatible with wiki username.
```

Comment:
For support requests, it's the admin - not the user - who sets wikiusername and whose savvy should provide the consistency and disallow confusion.  As an admin, I'd rather my users post on the wiki using the same name as the forums, or as close as possible where username restrictions allow.  The installation of Auth_phpBB implies this desire to tightly bind identity.